pkg/controller/connectivitycheck/connectivity_check_controller.go

```diff
@@ -158,25 +161,25 @@ func (c *connectivityCheckTemplateProvider) generate(ctx context.Context, syncCo
 			}
 			nodes[node.Name] = node
 		}
-		for _, template := range templates {
-			check := template.DeepCopy()
-			WithSource("network-check-source-" + strings.Split(pod.Spec.NodeName, ".")[0])(check)
-			check.Spec.SourcePod = pod.Name
-			nodeRef := metav1.OwnerReference{
-				APIVersion: "v1",
-				Kind:       "Node",
-				UID:        node.GetUID(),
-				Name:       node.GetName(),
+		for i := range checks {
+			WithSource("network-check-source-" + strings.Split(pod.Spec.NodeName, ".")[0])(checks[i])
+			checks[i].Spec.SourcePod = &pod.Name
```

Comment:
I don't think that this is correct, `pod` is an iteration variable so all of the checks will point to the last pod name returned by `c.diagnosticsPodLister.List(labels.Set{"app": "network-check-source"}.AsSelector())`

Reply:
@kyrtapz This is still inside pods iteration loop and not even on another goroutine, so i don't think there would be a variable jump. do you see a problem ?

Reply:
Now I got the problem what you are referring to. will update with a fix.

Reply:
this code is a little confusing to me, it looks like for every pod all of the checks will be modified to point to that specific pod?

Reply:
Yes @kyrtapz , i guess this is fixed correctly now. please check.